rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/Diagnostics.java

```diff
@@ -95,15 +95,33 @@ public static Diagnostic translateDiagnostic(IConstructor d, ColumnMaps cm) {
         Diagnostic result = new Diagnostic();
         result.setSeverity(severityMap.get(d.getName()));
         result.setMessage(((IString) d.get("msg")).getValue());
-        result.setRange(Locations.toRange(getMessageLocation(d), cm));
+        Range range = Locations.toRange(getMessageLocation(d), cm);
+        result.setRange(range);
+
+        storeFixCommands(d, result, range);
         return result;
     }
 
+    private static void storeFixCommands(IConstructor d, Diagnostic result, Range range) {
```

Comment:
Nit: Maybe swap the order of this function and the one below (to keep the two versions of `translateDiagnostic` together)

Reply:
Actually, it seems these two versions of `translateDiagnostics` are fully identical clones, except for the type of `cm`. I realize it's not the point of this PR, but as the PR makes these clones bigger, I feel at least a bit justified to complain about it 😉

Concrete suggestion:

```java
public static Diagnostic translateDiagnostic(IConstructor d, ColumnMaps cm) {
  return translateDiagnostic(d, Locations.toRange(getMessageLocation(d), cm));
}

public static Diagnostic translateDiagnostic(IConstructor d, LineColumnOffsetMap cm) {
  return translateDiagnostic(d, Locations.toRange(getMessageLocation(d), cm));
}

private static Diagnostic translateDiagnostic(IConstructor d, Range r) {
  ... // content of current translateDiagnostic
}
```

(There might be better/conciser alternatives.)

Reply:
nice! done.